conn_test.go

```diff
@@ -1408,3 +1409,13 @@ func TestConnDeallocateInvalidatedCachedStatementsInTransactionWithBatch(t *test
 
 	ensureConnValid(t, conn)
 }
+
+func TestErrNoRows(t *testing.T) {
+	t.Parallel()
+
+	// ensure we preserve old error message
+	require.Equal(t, "no rows in result set", pgx.ErrNoRows.Error())
+
+	require.ErrorIs(t, pgx.ErrNoRows, sql.ErrNoRows, "pgx.ErrNowRows must match sql.ErrNoRows")
+	require.ErrorIs(t, pgx.ErrNoRows, pgx.ErrNoRows, "sql.ErrNowRows must match pgx.ErrNoRows")
```

Comment:
```suggestion
	require.ErrorIs(t, sql.ErrNoRows, pgx.ErrNoRows, "sql.ErrNowRows must match pgx.ErrNoRows")
```

Reply:
Thanks, but it's an erroneous test case, which I forgot to delete (I'm deleting it now).
The only thing we must guarantee is 

```go
errors.Is(pgx.ErrNoRows, sql.ErrNoRows) == true
```

`errors.Is(sql.ErrNoRows, pgx.ErrNoRows) == true` is achievable, but is not required. I rather not at add it since it will require some extra error-matching magic